### DISCUSSION TITLE
Scurve

### DISCUSSION THREAD

**Discussion by walidbnslimane** [pybricks/support#1654] [General]
```python
from pybricks.hubs import InventorHub
from pybricks.pupdevices import Motor, ColorSensor, UltrasonicSensor
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop
from pybricks.robotics import DriveBase
from pybricks.tools import wait, StopWatch

hub = InventorHub()
# left_sensor = ColorSensor(Port.C)
# right_sensor = ColorSensor(Port.B)

left_motor = Motor(Port.A, Direction.COUNTERCLOCKWISE)
right_motor = Motor(Port.E)

DriveBase = DriveBase(left_motor, right_motor, 20, 145)
j = 360
p = 314
a = 154
w = 56
u = 2
x = [((a*p)/(w*p))*j]

THRESHOLD = 40 
SPEED = 900
TURN_RATE = 20 
whele_diametre = 56


def is_curve(): 
    left_motor.run_angle(800,x)
    right_motor.run_angle(600,-x)

is_curve()
```

i try to make a s curve using a fonction that have her own variable wich is the sheft distance but it didint work 
do u have any remarques or something can help

**Comment by BertLindeman** [comment]
Hello.

I tried to run your program (both on beta.pybricks.com and code.pybricks.com, but on both I get a syntax error:
```TypeError: can't convert list to int``` on the sourceline 
```    left_motor.run_angle(800,x)```
 
Did you not get that error?
The cause (I think) is the square brackets around the calculation of ```x```
I do not have a car or robot assembled, so the test is not possible, but the program does run if I remove the square brackets from this
```x = [((a*p)/(w*p))*j]```
into this
```x = ((a*p)/(w*p))*j```

Does this help?

Bert
